src/content/en/updates/2018/10/animation-worklet.md

```diff
@@ -0,0 +1,493 @@
+`localTime`, hence why this animator is called "passthrough". With this code for
+the worklet, the WAAPI vs AnimationWorklet comparison above behave exactly the
+same, as you can see in the
+[demo](https://animation-worklet-article.glitch.me/index.html).
+
+## Master of time
+
+The `currentTime` parameter of our `animate` method is the `currentTime` of the
+timeline we passed to the `WorkletTimeline` constructor. In the previous
+example, we just passed that time through to the effect. But since this is
+JavaScript code, and we can _distort time_ 💫
+
+    function remap(minIn, maxIn, minOut, maxOut, v) {
+      return (v - minIn)/(maxIn - minIn) * (maxOut - minOut) + minOut;
+    }
+    registerAnimator('sin', class {
+      animate(currentTime, effect) {
+        effect.localTime =
+          remap(-1, 1, 0, 2000, Math.sin(currentTime * 2 * Math.PI / 2000));
+      }
+    });
+
+Note: `currentTime` _can_ be `NaN` in certain circumstances (more later). Your
+should keep that in mind when writing animation worklet code. Since all
+mathematical operations can handle `NaN` (they return `NaN` when one of their
+inputs is `NaN`) we are fine here!
+
+We are taking the `Math.sin` of the `currentTime`, and remapping that value to
+the range [0; 2000], which is the time range that our effect is defined for. Now
+[the animation looks very
+different](https://animation-worklet-article.glitch.me/sin.html), without having
+changed the keyframes or the animation's options. The worklet code can be
+arbitrarily complex, and allows you to programmatically define which effects are
+played in which order and to which extent.
+
+<video loop muted autoplay
+  src="https://storage.googleapis.com/webfundamentals-assets/animation-worklet/sin_x264.mp4">
+</video>
+
+## Options over Options
+
+You might find yourself wanting to reuse a worklet and just wanting to adjust
+some numbers. For this reason the WorkletAnimation constructor allows you pass
+an options object to the worklet:
+
+    registerAnimator('factor', class {
+      constructor(options = {}) {
+        this.factor = options.factor || 1;
+      }
+      animate(currentTime, effect) {
+        effect.localTime = currentTime * this.factor;
+      }
+    });
+
+    new WorkletAnimation(
+      'factor',
+      new KeyframeEffect(
+        document.querySelector('#b'),
+        [ /* ... same keyframes as before ... */ ],
+        {
+          duration: 2000,
+          iterations: Number.POSITIVE_INFINITY
+        }
+      ),
+      document.timeline,
+      {factor: 0.5}
+    ).play();
+
+Note: The options object will be structurally cloned when it is being sent to
+the worklet, similar to how `postMessage()` operates.
+
+In this [example](https://animation-worklet-article.glitch.me/options.html),
+both animations are driven with the same worklet, but with different options.
+
+## Gimme your local state!
+
+As I hinted at before, one of the key problems AnimationWorklet aims to solve is
```

Comment:
Here's the rules on capitalizing and formatting API names.
* Naming generically: 'animation worklets'
* Naming the API formally: 'the Animation Worklet API'
* As an object name: `AnimationWorklet`.

This use is like the first. Remember to change the verb. Please check throughout.